fortanix-sgx-tools/src/bin/ftxsgx-runner.rs

```diff
@@ -7,38 +7,61 @@
 extern crate aesm_client;
 extern crate enclave_runner;
 extern crate sgxs_loaders;
-#[macro_use]
 extern crate failure;
-
-use std::path::Path;
+#[macro_use]
+extern crate clap;
 
 use aesm_client::AesmClient;
-use enclave_runner::Command;
+use enclave_runner::EnclaveBuilder;
 use failure::{Error, ResultExt};
 use sgxs_loaders::isgx::Device as IsgxDevice;
 
+use clap::{App, Arg};
+
+arg_enum!{
+    #[derive(PartialEq, Debug)]
+    #[allow(non_camel_case_types)]
+    pub enum Signature {
+        coresident,
+        dummy
+    }
+}
+
 fn main() -> Result<(), Error> {
-    let mut args = std::env::args_os();
-    let cmdname = args.next();
-    let file = match args.next() {
-        Some(arg) => arg,
-        None => {
-            let cmdname = cmdname
-                .as_ref()
-                .and_then(|p| Path::new(p).file_name())
-                .map(|p| Path::new(p).display().to_string());
-            eprintln!(
-                "Usage: {} <ENCLAVE>\n",
-                cmdname.unwrap_or("ftxsgx-runner".into())
-            ); // extra newline
-            bail!("Missing <ENCLAVE> parameter on command line");
-        }
-    };
+    let default_signature_value = format!("{:?}", &Signature::coresident);
+
+    let args = App::new("ftxsgx-runner")
+        .arg(
+            Arg::with_name("file")
+                .index(1)
```

Comment:
I think it might not be necessary to call `index`

Reply:
True, removed.